src/NuGetGallery/Controllers/UsersController.cs

```
@@ -182,13 +182,20 @@ public virtual ActionResult ForgotPassword()
 
             if (ModelState.IsValid)
             {
-                var user = await _authService.GeneratePasswordResetToken(model.Email, Constants.PasswordResetTokenExpirationHours * 60);
-                if (user != null)
+                try
                 {
-                    return SendPasswordResetEmail(user, forgotPassword: true);
-                }
+                    var user = await _authService.GeneratePasswordResetToken(model.Email, Constants.PasswordResetTokenExpirationHours * 60);
+                    if (user != null)
+                    {
+                        return SendPasswordResetEmail(user, forgotPassword: true);
+                    }
 
-                ModelState.AddModelError("Email", Strings.CouldNotFindAnyoneWithThatEmail);
+                    ModelState.AddModelError("Email", Strings.CouldNotFindAnyoneWithThatEmail);
+                }
+                catch (Exception e)
```

you shouldn't catch all exceptions.. catch UserSafeException explicitly..